_episodes/11-dask-configuration.md

```diff
@@ -0,0 +1,390 @@
+but if you are interested there is a
+[Dask Array Tutorial](https://tutorial.dask.org/02_array.html) as a well as a
+[guide to "Lazy Data"](https://scitools-iris.readthedocs.io/
+en/stable/userguide/real_and_lazy_data.html)
+available. Lazy data is the term the Iris library uses for Dask Arrays.
+
+
+### Dask Workers
+The most important concept to understand when using Dask Arrays is the concept
+of a Dask *worker*. With Dask, computations are run in parallel by little
+Python programs that are called *workers*. These could be on running on the
+same machine that you are running ESMValTool on, or they could be on one or
+more other computers. Dask workers typically require 2 to 4 gigabytes (GiB) of
+memory (RAM) each. In order to avoid running out of memory, it is important
+to use only as many workers as your computer(s) have memory for. ESMValCore
+(or Dask) provide configuration files where you can configure the number of
+workers.
+
+Note that only array computations are run using Dask, so total runtime may not
+decrease as much as you might expect when you increase the number of Dask
+workers.
+
+### Dask Scheduler
+
+In order to distribute the computations over the workers, Dask makes use of a
+*scheduler*. There are two different schedulers available. The default
+scheduler can be good choice for smaller computations that can run
+on a single computer, while the scheduler provided by the Dask Distributed
+package is more suitable for larger computations.
+
+> ## On using ``max_parallel_tasks``
+>
+> In the config-user.yml file, there is a setting called ``max_parallel_tasks``.
+> Any variable or diagnostic script in the recipe is considered a 'task' in this
+> context and when settings this to a value larger than 1, these will be
+> processed in parallel on the computer running the ``esmvaltool`` command.
+>
+> With the Dask Distributed scheduler, all the tasks running in parallel
+> can use the same workers, but with the default scheduler each task will
+> start its own workers. If a recipe does not run with ``max_parallel_tasks``
+> set to a value larger than 1, try reducing the value or setting it to 1.
+> This is especially the case for recipes with high resolution data or many
+> datasets per variable.
+>
+{: .callout}
+
+## Starting a Dask distributed cluster
+
+The workers and the scheduler together are called a Dask "cluster".
+Let's start the the tutorial by configuring ESMValCore so it runs its
+computations on a cluster with just one worker.
+
+We use a text editor called ``nano`` to edit the configuration file:
+
+~~~bash
+  nano ~/.esmvaltool/dask.yml
+~~~
+
+Any other editor can be used, e.g. many systems have ``vi`` available.
+
+This file contains the settings for:
+
+- Starting a new cluster of Dask workers
+- Or alternatively: connecting to an existing cluster of Dask workers
+
+Add the following content to the file ``~/.esmvaltool/dask.yml``:
+
```

Comment:
Should they have a template dask.yml file already or download it from somewehere to start with? (like the config file)

Reply:
Usually the content is only a few lines that easily be copy/pasted, so I'm not sure if that makes sense here.

Reply:
It would be nice to have pre-filled versions of these configuration files for specific clusters such as Jasmin or Levante, that is something I intend to work on once ESMValGroup/ESMValCore#2448 (currently pending review by the user engagement team) is merged.